kernel/src/hil/flash.rs

```diff
@@ -107,7 +113,11 @@ pub trait HasClient<'a, C> {
 /// A page of writable persistent flash memory.
 pub trait Flash {
     /// Type of a single flash page for the given implementation.
-    type Page: AsMut<[u8]>;
+    type Page: AsMut<[u8]> + Default;
+
+    /// Called by nonvolatile storage component. Responsible for chip-specific
+    /// configuration to enable flash.
+    fn configure(&self);
```

Comment:
I see where this comes from, but I think we would rather avoid adding `configure` as its unclear who should call it and when, particularly when it takes no arguments.

I think it would be pretty straightforward to check for the `FlashState::Unconfigured` state in the sam4l driver and call `self.configure()` if needed.